### ISSUE TITLE
sigsev on installing gems on aarch64

### ISSUE THREAD

**Issue by cpuguy83** [jruby/jruby#5447]
### Environment

Provide at least:
* JRuby version (`jruby -v`) and command line (flags, JRUBY_OPTS, etc)
9.2.*, 9.1.*
* Operating system and platform (e.g. `uname -a`)
unknown atm (build server not in my control)
* Using openjdk-8

### Expected Behavior

`gem install` should not crash

### Actual Behavior

When building jruby images for Docker, upon installing some base gems we hit a sigsev.
This is happening only on armv8 (aarch64) on alpine linux (which uses musl-libc).

```
Step 8/13 : RUN gem install bundler rake net-telnet xmlrpc
 ---> Running in ac6b889e4615
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x0000000000005e30, pid=1, tid=0x0000ffff86683a78
#
# JRE version: OpenJDK Runtime Environment (8.0_171-b11) (build 1.8.0_171-b11)
# Java VM: OpenJDK 64-Bit Server VM (25.171-b11 mixed mode linux-aarch64 compressed oops)
# Derivative: IcedTea 3.8.0
# Distribution: Custom build (Wed Jun 13 18:28:34 UTC 2018)
# Problematic frame:
# C  0x0000000000005e30
#
# Core dump written. Default location: //core or core.1
#
# An error report file with more information is saved as:
# //hs_err_pid1.log
#
# If you would like to submit a bug report, please include
# instructions on how to reproduce the bug and visit:
#   http://icedtea.classpath.org/bugzilla
#
```

**Comment by headius** [comment]
There should be a longer file "hs_err" something. Can you post that and link here?

**Comment by cpuguy83** [comment]
Here we go: https://gist.github.com/cpuguy83/508c7a0d39d7a0adf9a55bc69779e963

**Comment by headius** [comment]
So this is failing on 9.2.x and 9.1.x for you? I am unable to reproduce with JRuby master (9.2.5) on Ubuntu AArch64, so I suspect this is a musl-libc incompatibility.

**Comment by headius** [comment]
Well I do get an error on Alpine but it's not the segv. I'll keep poking around.

**Comment by headius** [comment]
Note: that's Alpine on x86_64.

**Comment by headius** [comment]
Ok, so the problem here is that musl-libc Linuxes don't actually provide a libcrypt since that functionality is part of musl-libc. When we attempt to load our native support, that binding of `crypt` fails and so we back off native.

I'm thinking we fix this and then have you test again, since it may solve the segv as well.

**Comment by headius** [comment]
I suspect the problem here was related to the failure to load libcrypt that I fixed in #123. The scenario I'm thinking is that perhaps the failure of the library to load was masked somehow on aarch64 and a partially bound jnr-posix was returned to the caller. With crypt bound to nothing, invoking it triggered a segv.

Anyway...here's hoping 😁
